gapic-common/lib/gapic/rest/fiber_enumerable.rb

```diff
@@ -0,0 +1,39 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Gapic
+  module Rest
+    class FiberEnumerable
```

Comment:
It doesn't look like you're using this enumerable for anything other than calling `next` on it. In which case, it seems like an unnecessary abstraction. You could just pass the fiber directly into the ServerStream.

Reply:
I want ServerStream to take a generic enumerable so that it's more testable. FiberEnumerable then wrap the handling of a Faraday call-wrapping Fiber

Reply:
Okay.